x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -161,6 +161,12 @@ export type ExternalCallbacksStorage = Map<ExternalCallback[0], Set<ExternalCall
  * Describes public Fleet plugin contract returned at the `startup` stage.
  */
 export interface FleetStartContract {
+  /**
+   * returns a promise that resolved when fleet setup has been completed regardless if it was successful or failed).
+   * Any consumer of fleet start services should first `await` for this promise to be resolved before using those
+   * services
+   */
+  fleetSetupCompleted: () => Promise<void>;
```

Comment:
## Why we need this

Fleet is a dependency of other plugins and the create/update/migrate of `.fleet-*` indexes is currently done during the Plugin's `start()` phase. Because of this, the services exposed via `FleetStartContract` should really only be used after Fleet has reported that its setup is done.

I hit a problem because we (endpoint) were attempting to use the `ArtifactClient` from fleet prior to the migration to `.fleet-*` indexes having been done.